docs/src/main/asciidoc/security-built-in-authentication-support-concept.adoc

```diff
@@ -187,7 +187,40 @@ public class HelloService {
 
 === How to customize authentication exception responses
 
-By default, the authentication security constraints are enforced before the JAX-RS chain starts and only way to handle Quarkus Security authentication exceptions is to provide a failure handler like this one:
+You can use JAX-RS `ExceptionMapper` to capture Quarkus Security authentication exceptions such as `io.quarkus.security.AuthenticationFailedException`, for example:
+
+[source,java]
+----
+package io.quarkus.it.keycloak;
+
+import javax.annotation.Priority;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+import io.quarkus.security.AuthenticationFailedException;
+
+@Provider
+@Priority(Priorities.AUTHENTICATION)
+public class AuthenticationFailedExceptionMapper implements ExceptionMapper<AuthenticationFailedException> {
+
+    @Context
+    UriInfo uriInfo;
+
+    @Override
+    public Response toResponse(AuthenticationFailedException exception) {
+        return Response.status(401).header("WWW-Authenticate", "Basic realm=\"Quarkus\"").build();
+    }
+}
+----
+
+CAUTION: By default, authentication mechanisms like `io.quarkus.oidc.runtime.OidcAuthenticationMechanism` needs to handle Quarkus Security authentication exceptions themselves in order redirect request and set correct response status.
+For that reason, built-in exception mappers that handles authentication failures challenge authentication mechanisms.
+See xref:security-customization.adoc#dealing-with-more-than-one-http-auth-mechanisms[Dealing with more than one HttpAuthenticationMechanism] section for more details on authentication mechanism challenge and more specifically {quarkus-blob-url}/extensions/resteasy-reactive/quarkus-resteasy-reactive/runtime/src/main/java/io/quarkus/resteasy/reactive/server/runtime/exceptionmappers/SecurityExceptionMapperUtil.java[SecurityExceptionMapperUtil.java] for an example how RESTEasy Reactive sends a challenge.
```

Comment:
```suggestion
In such cases, a safer way to customize authentication exceptions is to make sure the proactive authentication is not disabled and use Vert.x HTTP route failure handlers, as events come to the handler with the correct response status and headers.
```

Reply:
I thought the last sentence was too complex, with a link to the source is likely to be too technical kind of info for the docs, so propose just to replace it with a simple advice

Reply:
I think suggested way is clearer thanks.